Failed to build with Fedora build flags

When I am trying to build the package with Fedora macros, following error shows up.  My SPEC file is
[kabi-dw.spec](https://github.com/skozina/kabi-dw/files/1719837/kabi-dw.spec.txt) and I am building on Fedora 27 x86_64 with 

$ rpm -q rpm-build
rpm-build-4.14.1-1.fc27.x86_64
$ rpm -q redhat-rpm-config
redhat-rpm-config-70-1.fc27.noarch

```
Executing(%build): /bin/sh -e /var/tmp/rpm-tmp.ZMISWD
+ umask 022
+ cd /home/zsun/rpmbuild/BUILD
+ cd kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b
+ CFLAGS='-O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -m64 -mtune=generic'
+ LDFLAGS='-Wl,-z,relro -specs=/usr/lib/rpm/redhat/redhat-hardened-ld'
+ make debug
cc -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -m64 -mtune=generic -D_GNU_SOURCE -MM generate.c ksymtab.c utils.c main.c stack.c objects.c hash.c compare.c show.c > ./.depend
cc -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -m64 -mtune=generic -D_GNU_SOURCE -g -DDEBUG generate.c -o generate.o
/usr/lib/gcc/x86_64-redhat-linux/7/../../../../lib64/crt1.o: In function `_start':
(.text+0x20): undefined reference to `main'
/tmp/ccEg6Sor.o: In function `parse_generate_opts':
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1911: undefined reference to `rec_mkdir'
/tmp/ccEg6Sor.o: In function `record_free_regular':
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:467: undefined reference to `stack_pop'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:469: undefined reference to `stack_destroy'
/tmp/ccEg6Sor.o: In function `get_die_name':
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:192: undefined reference to `dwarf_hasattr'
/tmp/ccEg6Sor.o: In function `is_external':
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:384: undefined reference to `dwarf_hasattr'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:385: undefined reference to `dwarf_attr'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:386: undefined reference to `dwarf_hasform'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:391: undefined reference to `dwarf_attr'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:394: undefined reference to `dwarf_formref_die'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:395: undefined reference to `dwarf_diename'
/tmp/ccEg6Sor.o: In function `get_line':
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:286: undefined reference to `dwarf_attr'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:287: undefined reference to `dwarf_formudata'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:291: undefined reference to `dwarf_attr'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:292: undefined reference to `dwarf_formref_die'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:293: undefined reference to `dwarf_diename'
/tmp/ccEg6Sor.o: In function `die_read_alignment':
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:405: undefined reference to `dwarf_attr'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:408: undefined reference to `dwarf_formudata'
/tmp/ccEg6Sor.o: In function `_print_die_array_type':
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1171: undefined reference to `dwarf_tag'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1176: undefined reference to `dwarf_hasattr'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1177: undefined reference to `dwarf_attr'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1178: undefined reference to `dwarf_formudata'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1189: undefined reference to `dwarf_siblingof'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1193: undefined reference to `obj_array_new_add'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1181: undefined reference to `dwarf_hasattr'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1182: undefined reference to `dwarf_attr'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1183: undefined reference to `dwarf_formudata'
/tmp/ccEg6Sor.o: In function `record_dump':
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:753: undefined reference to `rec_mkdir'
/tmp/ccEg6Sor.o: In function `record_dump_weak':
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:732: undefined reference to `filenametosymbol'
/tmp/ccEg6Sor.o: In function `record_dump_assembly':
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:721: undefined reference to `filenametosymbol'
/tmp/ccEg6Sor.o: In function `_get_file':
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:225: undefined reference to `dwarf_attr'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:229: undefined reference to `dwarf_formudata'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:231: undefined reference to `dwarf_getsrcfiles'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:234: undefined reference to `dwarf_filesrc'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:248: undefined reference to `path_normalize'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:232: undefined reference to `dwarf_diename'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:226: undefined reference to `dwarf_diename'
/tmp/ccEg6Sor.o: In function `hash_find':
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/hash.h:47: undefined reference to `hash_find_bin'
/tmp/ccEg6Sor.o: In function `record_merge':
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:808: undefined reference to `obj_merge'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:812: undefined reference to `obj_fill_parent'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:814: undefined reference to `obj_free'
/tmp/ccEg6Sor.o: In function `hash_add_unique':
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/hash.h:37: undefined reference to `hash_add_unique_bin'
/tmp/ccEg6Sor.o: In function `ksymtab_add_and_link_alias':
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1625: undefined reference to `ksymtab_find'
/tmp/ccEg6Sor.o: In function `record_stack_dump_and_clear':
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:684: undefined reference to `stack_pop'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:693: undefined reference to `stack_pop'
/tmp/ccEg6Sor.o: In function `is_declaration':
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:206: undefined reference to `dwarf_hasattr'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:208: undefined reference to `dwarf_attr'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:209: undefined reference to `dwarf_hasform'
/tmp/ccEg6Sor.o: In function `process_symbol_file':
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1666: undefined reference to `ksymtab_read'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1668: undefined reference to `ksymtab_len'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1688: undefined reference to `ksymtab_free'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1689: undefined reference to `ksymtab_free'
/tmp/ccEg6Sor.o: In function `merge_aliases':
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1640: undefined reference to `ksymtab_for_each'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1642: undefined reference to `ksymtab_for_each'
/tmp/ccEg6Sor.o: In function `generate_type_info':
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1516: undefined reference to `dwfl_begin'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1518: undefined reference to `dwfl_report_offline'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1522: undefined reference to `dwfl_report_end'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1523: undefined reference to `dwfl_getmodules'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1525: undefined reference to `dwfl_end'
/tmp/ccEg6Sor.o: In function `process_symbol_file':
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1683: undefined reference to `ksymtab_for_each'
/tmp/ccEg6Sor.o: In function `is_all_done':
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1533: undefined reference to `ksymtab_mark_count'
/tmp/ccEg6Sor.o: In function `generate_type_info':
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1519: undefined reference to `dwfl_report_end'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1520: undefined reference to `dwfl_errmsg'
/tmp/ccEg6Sor.o: In function `process_not_found':
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1600: undefined reference to `ksymtab_find'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1603: undefined reference to `ksymtab_ksym_mark'
/tmp/ccEg6Sor.o: In function `print_subprogram_arguments':
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1105: undefined reference to `dwarf_haschildren'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1109: undefined reference to `dwarf_child'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1121: undefined reference to `obj_var_new_add'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1125: undefined reference to `obj_list_add'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1127: undefined reference to `dwarf_siblingof'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1112: undefined reference to `dwarf_tag'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1113: undefined reference to `dwarf_tag'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1116: undefined reference to `dwarf_tag'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1119: undefined reference to `obj_basetype_new'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1123: undefined reference to `obj_list_head_new'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1127: undefined reference to `dwarf_siblingof'
/tmp/ccEg6Sor.o: In function `print_die_subprogram':
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1146: undefined reference to `obj_func_new_add'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1146: undefined reference to `obj_func_new_add'
/tmp/ccEg6Sor.o: In function `print_die_tag':
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1222: undefined reference to `dwarf_tag'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1223: undefined reference to `dwarf_diename'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1235: undefined reference to `obj_var_new_add'
/tmp/ccEg6Sor.o: In function `print_die_array_type':
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1207: undefined reference to `dwarf_haschildren'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1213: undefined reference to `dwarf_child'
/tmp/ccEg6Sor.o: In function `print_die_enumeration':
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1029: undefined reference to `obj_enum_new'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1031: undefined reference to `dwarf_haschildren'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1034: undefined reference to `dwarf_child'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1043: undefined reference to `obj_list_add'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1044: undefined reference to `dwarf_siblingof'
/tmp/ccEg6Sor.o: In function `print_die_enumerator':
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1009: undefined reference to `dwarf_attr'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1012: undefined reference to `dwarf_formudata'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1014: undefined reference to `obj_constant_new'
/tmp/ccEg6Sor.o: In function `print_die_enumeration':
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1041: undefined reference to `obj_list_head_new'
/tmp/ccEg6Sor.o: In function `print_die_tag':
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1242: undefined reference to `obj_ptr_new_add'
/tmp/ccEg6Sor.o: In function `print_die_structure':
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:972: undefined reference to `obj_struct_new'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:974: undefined reference to `dwarf_haschildren'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:977: undefined reference to `dwarf_child'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:991: undefined reference to `obj_list_add'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:993: undefined reference to `dwarf_siblingof'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:982: undefined reference to `dwarf_tag'
/tmp/ccEg6Sor.o: In function `print_die_struct_member':
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:930: undefined reference to `dwarf_attr'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:933: undefined reference to `dwarf_formudata'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:936: undefined reference to `obj_struct_member_new_add'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:939: undefined reference to `dwarf_hasattr'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:942: undefined reference to `dwarf_hasattr'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:946: undefined reference to `dwarf_attr'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:948: undefined reference to `dwarf_formudata'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:949: undefined reference to `dwarf_attr'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:951: undefined reference to `dwarf_formudata'
/tmp/ccEg6Sor.o: In function `print_die_structure':
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:989: undefined reference to `obj_list_head_new'
/tmp/ccEg6Sor.o: In function `print_die_tag':
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1262: undefined reference to `obj_qualifier_new_add'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1255: undefined reference to `obj_typedef_new_add'
/tmp/ccEg6Sor.o: In function `print_die_union':
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1064: undefined reference to `obj_union_new'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1066: undefined reference to `dwarf_haschildren'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1069: undefined reference to `dwarf_child'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1085: undefined reference to `obj_list_add'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1087: undefined reference to `dwarf_siblingof'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1074: undefined reference to `dwarf_tag'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1080: undefined reference to `obj_var_new_add'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1083: undefined reference to `obj_list_head_new'
/tmp/ccEg6Sor.o: In function `print_die_tag':
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1238: undefined reference to `obj_basetype_new'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1267: undefined reference to `obj_qualifier_new_add'
/tmp/ccEg6Sor.o: In function `get_symbol_file':
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:327: undefined reference to `dwarf_diename'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:328: undefined reference to `dwarf_tag'
/tmp/ccEg6Sor.o: In function `hash_find':
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/hash.h:47: undefined reference to `hash_find_bin'
/tmp/ccEg6Sor.o: In function `hash_add':
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/hash.h:31: undefined reference to `hash_add_bin'
/tmp/ccEg6Sor.o: In function `print_die':
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1336: undefined reference to `obj_reffile_new'
/tmp/ccEg6Sor.o: In function `record_new_regular':
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:509: undefined reference to `stack_init'
/tmp/ccEg6Sor.o: In function `record_add_stack':
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:588: undefined reference to `walk_stack_backward'
/tmp/ccEg6Sor.o: In function `print_die':
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1321: undefined reference to `stack_push'
/tmp/ccEg6Sor.o: In function `record_close':
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:678: undefined reference to `obj_fill_parent'
/tmp/ccEg6Sor.o: In function `print_die':
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1336: undefined reference to `obj_reffile_new'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1324: undefined reference to `stack_pop'
/tmp/ccEg6Sor.o: In function `get_file':
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:265: undefined reference to `dwarf_hasattr'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:268: undefined reference to `dwarf_attr'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:269: undefined reference to `dwarf_formref_die'
/tmp/ccEg6Sor.o: In function `record_add_cu':
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:598: undefined reference to `dwarf_diename'
/tmp/ccEg6Sor.o: In function `get_file':
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:270: undefined reference to `dwarf_diename'
/tmp/ccEg6Sor.o: In function `print_die_type':
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:908: undefined reference to `dwarf_hasattr'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:909: undefined reference to `obj_basetype_new'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:911: undefined reference to `dwarf_attr'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:912: undefined reference to `dwarf_formref_die'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:913: undefined reference to `dwarf_diename'
/tmp/ccEg6Sor.o: In function `dwflmod_generate_cb':
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1475: undefined reference to `dwfl_module_getdwarf'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1491: undefined reference to `dwarf_next_unit'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1498: undefined reference to `dwarf_offdie'
/tmp/ccEg6Sor.o: In function `process_cu_die':
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1432: undefined reference to `dwarf_haschildren'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1436: undefined reference to `dwarf_child'
/tmp/ccEg6Sor.o: In function `is_symbol_valid':
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1360: undefined reference to `ksymtab_find'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1366: undefined reference to `ksymtab_find'
/tmp/ccEg6Sor.o: In function `process_cu_die':
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1468: undefined reference to `dwarf_siblingof'
/tmp/ccEg6Sor.o: In function `is_symbol_valid':
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1347: undefined reference to `dwarf_diename'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1348: undefined reference to `dwarf_tag'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1379: undefined reference to `ksymtab_ksym_mark'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1415: undefined reference to `ksymtab_ksym_mark'
/tmp/ccEg6Sor.o: In function `process_cu_die':
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1454: undefined reference to `stack_init'
/tmp/ccEg6Sor.o: In function `set_init':
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:418: undefined reference to `hash_new'
/tmp/ccEg6Sor.o: In function `process_cu_die':
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1460: undefined reference to `obj_free'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1463: undefined reference to `stack_pop'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1466: undefined reference to `stack_destroy'
/tmp/ccEg6Sor.o: In function `set_free':
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:447: undefined reference to `hash_free'
/tmp/ccEg6Sor.o: In function `process_cu_die':
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1445: undefined reference to `dwarf_diename'
/tmp/ccEg6Sor.o: In function `set_init':
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:418: undefined reference to `hash_new'
/tmp/ccEg6Sor.o: In function `read_symbols':
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1808: undefined reference to `ksymtab_new'
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1824: undefined reference to `ksymtab_add_sym'
/tmp/ccEg6Sor.o: In function `generate':
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1923: undefined reference to `ksymtab_len'
/tmp/ccEg6Sor.o: In function `record_db_init':
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:876: undefined reference to `hash_new'
/tmp/ccEg6Sor.o: In function `generate_symbol_defs':
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1729: undefined reference to `ksymtab_for_each'
/tmp/ccEg6Sor.o: In function `record_db_dump':
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:889: undefined reference to `hash_iter_init'
/tmp/ccEg6Sor.o: In function `hash_iter_next':
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/hash.h:52: undefined reference to `hash_iter_next_bin'
/tmp/ccEg6Sor.o: In function `record_db_free':
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:898: undefined reference to `hash_free'
/tmp/ccEg6Sor.o: In function `generate':
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1932: undefined reference to `ksymtab_free'
/tmp/ccEg6Sor.o: In function `generate_symbol_defs':
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1720: undefined reference to `walk_dir'
/tmp/ccEg6Sor.o: In function `ksymtab_add_alias':
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1614: undefined reference to `ksymtab_copy_sym'
/tmp/ccEg6Sor.o: In function `record_free_regular':
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:471: undefined reference to `obj_free'
/tmp/ccEg6Sor.o: In function `get_die_name':
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:193: undefined reference to `dwarf_diename'
/tmp/ccEg6Sor.o: In function `copy_stack_cb':
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:583: undefined reference to `stack_push'
/tmp/ccEg6Sor.o: In function `ksymtab_add_and_link_alias':
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:1633: undefined reference to `ksymtab_copy_sym'
/tmp/ccEg6Sor.o: In function `record_dump_regular':
/home/zsun/rpmbuild/BUILD/kabi-dw-05c38df0dfc2e0e3eeb605a34234a1bf77c7ea7b/generate.c:716: undefined reference to `obj_dump'
/tmp/ccEg6Sor.o:(.data.rel.ro+0x8): undefined reference to `dwfl_standard_find_debuginfo'
/tmp/ccEg6Sor.o:(.data.rel.ro+0x10): undefined reference to `dwfl_offline_section_address'
collect2: error: ld returned 1 exit status
make: *** [Makefile:71: generate.o] Error 1
error: Bad exit status from /var/tmp/rpm-tmp.ZMISWD (%build)


RPM build errors:
    Bad exit status from /var/tmp/rpm-tmp.ZMISWD (%build)
```

--- a/Makefile
+++ b/Makefile
@@ -18,8 +18,8 @@ SRCS=generate.c ksymtab.c utils.c main.c stack.c objects.c hash.c
 SRCS += compare.c show.c
 
 CC?=gcc
-CFLAGS?=-Wall -O2 --std=gnu99 -D_GNU_SOURCE -c
-LDFLAGS?=-ldw -lelf
+CFLAGS+=-Wall -O2 --std=gnu99 -D_GNU_SOURCE -c
+LDFLAGS+=-ldw -lelf
 
 YACC=bison
 YACCFLAGS=-d -t

--- a/README.md
+++ b/README.md
@@ -58,12 +58,12 @@ Linux has an option (CONFIG\_MODVERSIONS) to generate a checksum identifing all 
 
 This program needs elfutils installed. Check out your distribution to figure out how to install elfutils.
 
-For *Fedora* systems:
+For *Fedora* and *CentOS 7* systems:
 ~~~
-dnf install elfutils-devel
+dnf install kabi-dw
 ~~~
 
-For *Ubuntu* systems:
+For *Ubuntu* systems, you need to compile it. Install the dependencies:
 ~~~
 sudo apt-get install elfutils
 ~~~

--- a/utils.c
+++ b/utils.c
@@ -141,7 +141,7 @@ int check_is_directory(char *dir)
 static void safe_mkdir(char *path)
 {
 	if (mkdir(path, S_IRWXU | S_IRGRP | S_IXGRP | S_IROTH | S_IXOTH) != 0)
-		fail(strerror(errno));
+		fail("%s", strerror(errno));
 }
 
 void rec_mkdir(char *path)
@@ -175,7 +175,7 @@ void rec_mkdir(char *path)
 			if (rv == ENOENT)
 				safe_mkdir(buf);
 			else
-				fail(strerror(rv));
+				fail("%s", strerror(rv));
 		}
 
 		if (pos != NULL)

